packages/atclient/src/connection.c

```diff
@@ -757,6 +758,7 @@ static int atclient_connection_set_host(atclient_connection *ctx, const char *ho
     atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "Failed to allocate memory for host\n");
     goto exit;
   }
+  free(old_host);
```

Comment:
We only want to free this piece of memory if the allocation for the new host succeeds